frontend/packages/dev-console/src/components/helm/HelmReleaseUpgradePage.tsx

```diff
@@ -0,0 +1,100 @@
+import HelmActionPage from './HelmActionPage';
+
+type HelmReleaseUpgradePageProps = RouteComponentProps<{ ns: string; releaseName: string }>;
+
+export const HelmReleaseUpgradePage: React.FC<HelmReleaseUpgradePageProps> = (props) => {
+  const {
+    match: {
+      params: { ns: namespace, releaseName },
+    },
+  } = props;
+
+  const [chartValues, setChartValues] = React.useState({});
+  const [chartName, setChartName] = React.useState('');
+  const [helmChartData, setHelmChartData] = React.useState<HelmChartMetaData[]>([]);
+  const [chartValuesLoaded, setChartValuesLoaded] = React.useState(false);
+  const [chartVersions, setChartVersions] = React.useState({});
+  const [activeChartVersion, setActiveChartVersion] = React.useState<string>('');
+
+  React.useEffect(() => {
+    const getChartDetails = () => {
+      coFetchJSON(`/api/helm/release?ns=${namespace}&release_name=${releaseName}`)
+        .then((res) => {
+          setActiveChartVersion(res.chart.metadata.version);
+          setChartName(res.chart.metadata.name);
+          setChartValues(res.chart.values);
+          setChartValuesLoaded(true);
+        })
+        .catch((err) => {
+          // eslint-disable-next-line no-console
+          console.error(err);
+        });
+    };
+
+    getChartDetails();
+  }, [namespace, releaseName]);
+
+  React.useEffect(() => {
+    const getChartVersions = () => {
+      let json: HelmRelease;
+
+      coFetch('/api/helm/charts/index.yaml')
+        .then(async (response) => {
+          const yaml = await response.text();
+          json = safeLoad(yaml);
+          const versions = _.reduce(
+            _.get(json, ['entries', chartName]),
+            (obj, chart) => {
+              obj[chart.version] = chart.version;
+              return obj;
+            },
+            {},
+          );
+          setHelmChartData(_.get(json, ['entries', chartName]));
+          setChartVersions(versions);
+        })
+        .catch((err) => {
+          // eslint-disable-next-line no-console
+          console.error(err);
+        });
+    };
+    getChartVersions();
+  }, [chartName]);
+
+  const onVersionChange = (value: string) => {
+    setActiveChartVersion(value);
+    return getChartURL(helmChartData, value);
+  };
+
+  const helmUpgradeRequest = (payload: object) => coFetchJSON.put('/api/helm/release', payload);
+
+  if (!chartValuesLoaded) {
+    return <LoadingBox />;
+  }
+  return (
+    <HelmActionPage
+      namespace={namespace}
+      releaseName={releaseName}
+      title={'Upgrade Helm Release'}
+      helmAction={HelmActionType.Upgrade}
+      chartName={chartName}
+      chartVersions={chartVersions}
+      activeChartVersion={activeChartVersion}
+      historyURL={`/helm-releases/ns/${namespace}`}
+      chartValues={chartValues}
+      onChartVersionChange={onVersionChange}
+      helmActionRequest={helmUpgradeRequest}
+      chartURL={getChartURL(helmChartData, activeChartVersion)}
```

Comment:
Why is this needed to be passed from here? I though the backend is now supporting request without `chartURL`.

Reply:
removed it